functions/api/dashboard.js

```diff
@@ -0,0 +1,124 @@
+const constants = require('../common/constants')
+const { getInventory } = require('./inventory')
+const admin = require('firebase-admin');
+const express = require('express');
+const router = express.Router();
+const db = admin.firestore();
+
+/**
+ * @description Route to retireve all data for the dashboard from firestore
+ * If the user is Admin or SuperAdmin return all dashboard data from all Branches
+ * If the user is Branch Userreturn dashboard data from that specific branch
+ * @returns Json object containing all metadata
+ */
+router.get("/", async (request, response, next) => {
+
+    const userRole = request.user.role
+    console.debug(`Retrieving Dashboard data for ${userRole.toUpperCase()}`)
+    if (userRole === constants.ADMIN || userRole === constants.SUPER_ADMIN) {
+
+        //Get dashboard data for all branches
+        const dashboardDataForAllBranches = {}
+        const dashboardData = await getDashboardDataForAdminUser()
+        dashboardData.forEach(dashboard => {
+            const branchId = dashboard[constants.BRANCH]
+            delete dashboard[constants.BRANCH]
```

Comment:
why are we deleting the branch?

Reply:
I have reused the old methods but Vinayak wants it in different format. So, I had to delete the branch.